pkg/controller/datavolume-controller.go

```diff
@@ -1236,6 +1243,75 @@ func (r *DatavolumeReconciler) newPersistentVolumeClaim(dataVolume *cdiv1.DataVo
 				}),
 			},
 		},
-		Spec: *dataVolume.Spec.PVC,
+		Spec: *pvcSpec,
 	}, nil
 }
+
+func (r *DatavolumeReconciler) renderPvcSpec(dv *cdiv1.DataVolume) (*corev1.PersistentVolumeClaimSpec, error) {
+	// right now Spec.PVC is required and tested before, so no need to check if its null here!
+	pvcSpecCopy := dv.Spec.PVC.DeepCopy()
+	storageClass, err := GetStorageClassByName(r.client, dv.Spec.PVC.StorageClassName)
+	if err != nil {
+		return nil, err
+	}
+	if storageClass == nil {
+		// Not even default storageClass on the cluster, cannot apply the defaults
+		return pvcSpecCopy, nil
```

Comment:
right now in Kubernetes and cdi you can actually create a pvc without storageclass, it will be pending, until default storageclass is created. 

But maybe we do not want it that way. If I return error here, we have a simpler/cleaner logic, where storageclass (any or default) is always required. But it will change current behavior.

Is current behavior intentional?

Reply:
There is no requirement in k8s that a storage class exists. You can make PVs and PVCs without storage classes and if possible they will be bound. A good example of 'clusters' doing this is CRC or minikube where they create a bunch of PVs without any storage class. We still need to work with this case.